utils/providers.py

```diff
@@ -176,13 +177,24 @@ def setup_a_provider(
             # If our filtering yielded any providers, use them, otherwise do not bother with that
             providers = filtered_providers
 
+    # Keep track of providers that do not behave correctly
+    bad_providers = set([])
     # If there is already a suitable provider, don't try to setup a new one.
     already_existing = filter(is_provider_setup, providers)
-    if already_existing:
-        chosen = random.choice(already_existing)
+    random.shuffle(already_existing)
```

Comment:
This is going to significantly impact the already existing running tests isn't it? This setup_a_provider is called often I thought. Meaning that each time we want to setup_a_provider, we validate that all the existing ones are valid. Especially when people are running locally with 3+ providers, isn't this going to significantly impact them? Or do I have it wrong?